datatransfer/impl/graphsync/graphsync.go

```diff
@@ -106,7 +110,13 @@ func (impl *graphsyncImpl) OpenPullDataChannel(ctx context.Context, to peer.ID,
 	if err != nil {
 		return datatransfer.ChannelID{}, err
 	}
-	return datatransfer.ChannelID{To: to, ID: tid}, nil
+	chid := impl.createNewChannel(tid, to, baseCid, selector, voucher)
+	return chid, nil
+}
+
+// createNewChannel creates a new channel id
+func (impl *graphsyncImpl) createNewChannel(tid datatransfer.TransferID, to peer.ID, baseCid cid.Cid, selector ipld.Node, voucher datatransfer.Voucher) (datatransfer.ChannelID) {
```

Comment:
This is a temporary name. Later this func will also create a ChannelState and add it to the list of channels.